scripts/amd/gemm/README.md

```diff
@@ -1,44 +1,164 @@
+    - randn (default): normal distribution, i.e. `torch.randn`
+  - `--rotating_tensor SIZE`: provide the size of memory used for rotatin tensor.
+  The default is 0, meaning rotating tensor is not used.
+  - `--icahe_flush`: If true, the script will generate a kernel to flush i-cache.
+  The default is False.
+  - `--bias_vector`: If true, a bias vector along the M dim is applied.
+  The default is False.
+- Correctness check
+  - `--compare` will check the correctness of the best config for each gemm size.
+  - `--compare_wo_tuning` will check the correctness of the config provided in
+  the yaml file. If this is set, user needs to provide all the parameters in
+  the input yaml file. Example can be found in the benchmark mode section.
+- Logistics
+  - `--keep` can be used to keep the files generated during the tuning process.
+  Be default, intermediate files are removed at the end.
+  - `--time_breakdown`: If set, the script will print out elapsed time during
+  each stage of the tuning in real-time. The default is False.
+  - `--verbose` will enable more logging message than `--time_breakdown`, such
+  as output from rocprofv2
+  - `--o OUTPUT` can be used to control the output filename to store the tuning
+  result. The default filename is `tuning_results_branchName@gitCommit_timeStamp.yaml`.
+  Therefore, each time the user runs the tuning script, a different output file
+  will be generated.
+
+Here are some example usages of running the script for tuning:
+
+Tune some gemm sizes with f16 input
+```python
+./tune_gemm --gemm_size_file input.yaml --ngpus 8 --jobs 32 --o output.yaml
 ```
+It's recommended to use as many GPUs as possible and set `--jobs` to
+a value that is 4 to 6 times the number of GPUs.
 
-2. Tune a single gemm size
-```bash
-python tune_gemm.py -m 16 -n 16 -k 16
+If you are only allowed to use a subset of the GPUs, you can
+```python
+./tune_gemm --gemm_size_file input.yaml --gpu_ids 0,1,3,4 --jobs 32 --o output.yaml
 ```
+This runs the profiling on GPU 0,1,3,4.
 
-3. Choose the file to store tuning results
-```bash
-python tune_gemm.py --gemm_size_file input_gemm_sizes.yaml --o output_tuning.yaml
+For bf8 input
+```python
+./tune_gemm --gemm_size_file input.yaml --ngpus 8 --jobs 32 -dtype_a bf8 -dtype_b bf8
 ```
 
-4. Only check correctness given the tuning results
-```bash
-python tune_gemm.py --gemm_size_file output_tuning.yaml --compare_wo_tuning
+Check correctness of the tuned configs
+```python
+./tune_gemm --gemm_size_file output.yaml --compare_wo_tuning
 ```
-Note that the tuning results file are provided as the `gemm_size_file` in this scenario.
+
+
+### Benchmark mode
+
+In benchmark mode, the script will run a single given config multiple times to
+collect performance data. The benchmark mode works as
+The tuning script works as follows
+```python
+./tune_gemm --gemm_size_file input.yaml [options] --benchmark
+```
+The supported `options` are as followings
```

Comment:
So icache flush and rotating tensor are not supported in benchmark mode?

Reply:
They do, I forgot to add then here.